modules/mysql/variables.tf

```diff
@@ -16,151 +16,193 @@
 
 variable "project_id" {
   description = "The project ID to manage the Cloud SQL resources"
+  type        = string
 }
 
 variable "name" {
+  type        = string
   description = "The name of the Cloud SQL resources"
 }
 
 // required
 variable "database_version" {
   description = "The database version to use"
+  type        = string
 }
 
 // required
 variable "region" {
   description = "The region of the Cloud SQL resources"
+  type        = string
   default     = "us-central1"
 }
 
 // Master
 variable "tier" {
   description = "The tier for the master instance."
+  type        = string
   default     = "db-n1-standard-1"
 }
 
 variable "zone" {
   description = "The zone for the master instance, it should be something like: `a`, `c`."
+  type        = string
 }
 
 variable "activation_policy" {
   description = "The activation policy for the master instance. Can be either `ALWAYS`, `NEVER` or `ON_DEMAND`."
+  type        = string
   default     = "ALWAYS"
 }
 
 variable "authorized_gae_applications" {
   description = "The list of authorized App Engine project names"
+  type        = list(string)
   default     = []
 }
 
 variable "disk_autoresize" {
   description = "Configuration to increase storage size"
+  type        = bool
   default     = true
 }
 
 variable "disk_size" {
   description = "The disk size for the master instance"
+  type        = number
   default     = 10
 }
 
 variable "disk_type" {
   description = "The disk type for the master instance."
+  type        = string
   default     = "PD_SSD"
 }
 
 variable "pricing_plan" {
   description = "The pricing plan for the master instance."
+  type        = string
   default     = "PER_USE"
 }
 
 variable "maintenance_window_day" {
   description = "The day of week (1-7) for the master instance maintenance."
+  type        = number
   default     = 1
 }
 
 variable "maintenance_window_hour" {
   description = "The hour of day (0-23) maintenance window for the master instance maintenance."
+  type        = number
   default     = 23
 }
 
 variable "maintenance_window_update_track" {
   description = "The update track of maintenance window for the master instance maintenance. Can be either `canary` or `stable`."
+  type        = string
   default     = "canary"
 }
 
 variable "database_flags" {
-  description = "The database flags for the master instance. See [more details](https://cloud.google.com/sql/docs/mysql/flags)"
+  description = "List of Cloud SQL flags that are applied to the database server"
+  type        = list(map(string))
```

Comment:
Map is enough because of same type for all of the values:

						"database_flags": {
							Type:     schema.TypeList,
							Optional: true,
							Elem: &schema.Resource{
								Schema: map[string]*schema.Schema{
									"value": {
										Type:     schema.TypeString,
										Optional: true,
									},
									"name": {
										Type:     schema.TypeString,
										Optional: true,
									},
								},
							},